paymentsheet/src/main/java/com/stripe/android/paymentsheet/ConfirmCallback.kt

```diff
@@ -0,0 +1,55 @@
+package com.stripe.android.paymentsheet
+
+import androidx.annotation.RestrictTo
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface ConfirmCallback {
+
+    /**
+     * 🚧 Under construction 🚧
+     * The callback to implement to retrieve the intent client secret in the deferred flow.
+     *
+     * @param paymentMethodId the payment method ID to create the intent with
+     *
+     * @return a [ConfirmResponse] which contains the intent client secret and whether or not the
+     * intent was confirmed on the integrators server.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    suspend fun onRetrieveConfirmResponse(paymentMethodId: String): ConfirmResponse
+}
+
+internal fun interface LegacyConfirmCallback : ConfirmCallback {
+    override suspend fun onRetrieveConfirmResponse(paymentMethodId: String): ConfirmResponse {
+        return suspendCoroutine { continuation ->
+            onRetrieveConfirmResponse(
+                paymentMethodId,
+                { continuation.resume(it) },
+                { continuation.resumeWithException(it) }
+            )
+        }
+    }
+
+    fun onRetrieveConfirmResponse(
+        paymentMethodId: String,
+        onSuccess: (ConfirmResponse) -> Unit,
+        onError: (Throwable) -> Unit
+    )
+}
+
+/**
+ * 🚧 Under construction 🚧
+ * The model for the confirmation response from the integrators servers.
+ *
+ * @param clientSecret the intent client secret
+ * @param confirmedServerSide whether or not the intent was confirmed server side. If true, then
+ * the intent was confirmed on the integrators server. If false, the intent will be confirmed client
+ * side.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class ConfirmResponse(
```

Comment:
Can we model this as a result class? I’m imagining something like this:

```kotlin
interface ConfirmCallback {
  // …
  
  sealed interface Result {
    data class Success(val clientSecret: String) : Result
    data class Failure(val error: String) : Result
  }
}
```

And I don’t think we’ll expose a `confirmed` parameter. We’ll determine the confirmed state of the intent when we retrieve it on our end, as we need to inspect it anyway.